dbms/src/Functions/FunctionsJson.h

```diff
@@ -533,49 +533,305 @@ class FunctionJsonLength : public IFunction
+                data_to[row] = json_binary.getElementCount();
+            json_source->next();
         }
+
+        if constexpr (is_nullable)
+            return ColumnNullable::create(std::move(col_to), std::move(col_null_map));
         else
-            throw Exception(
-                fmt::format("Illegal column {} of argument of function {}", column->getName(), getName()),
-                ErrorCodes::ILLEGAL_COLUMN);
+            return col_to;
+    }
+
+    template <bool is_json_nullable, bool is_path_nullable>
+    static MutableColumnPtr doExecuteCommonForTwoArgs(
+        const std::unique_ptr<IStringSource> & json_source,
+        const NullMap & null_map_json,
+        const std::unique_ptr<IStringSource> & path_source,
+        const NullMap & null_map_path,
+        size_t rows)
+    {
+#define FINISH_PER_ROW   \
+    json_source->next(); \
+    path_source->next();
+
+        auto col_to = ColumnUInt64::create(rows, 1);
+        auto & data_to = col_to->getData();
+        ColumnUInt8::MutablePtr col_null_map = ColumnUInt8::create(rows, 0);
+        ColumnUInt8::Container & null_map_to = col_null_map->getData();
+        for (size_t row = 0; row < rows; ++row)
+        {
+            if constexpr (is_json_nullable)
+            {
+                if (null_map_json[row])
+                {
+                    FINISH_PER_ROW
+                    null_map_to[row] = 1;
+                    continue;
+                }
+            }
+
+            const auto & json_val = json_source->getWhole();
+            assert(json_val.size > 0);
+            JsonBinary json_binary(json_val.data[0], StringRef(&json_val.data[1], json_val.size - 1));
+            if (json_binary.getType() == JsonBinary::TYPE_CODE_ARRAY
+                || json_binary.getType() == JsonBinary::TYPE_CODE_OBJECT)
+            {
+                if constexpr (is_path_nullable)
+                {
+                    if (null_map_path[row])
+                    {
+                        FINISH_PER_ROW
+                        null_map_to[row] = 1;
+                        continue;
+                    }
+                }
+
+                const auto & path_val = path_source->getWhole();
+                auto path_expr_container_vec = buildPathExprContainer(StringRef{path_val.data, path_val.size});
+                auto extract_json_binaries = json_binary.extract(path_expr_container_vec);
+                if (extract_json_binaries.empty())
+                {
+                    FINISH_PER_ROW
+                    null_map_to[row] = 1;
+                    continue;
+                }
+                const auto & extract_json_binary = extract_json_binaries.back();
+                if (extract_json_binary.getType() == JsonBinary::TYPE_CODE_ARRAY
+                    || extract_json_binary.getType() == JsonBinary::TYPE_CODE_OBJECT)
+                    data_to[row] = extract_json_binary.getElementCount();
+            }
+            FINISH_PER_ROW
```

Comment:
If the current json is not json_array or json_object, no matter the path is, it will always return 1, is it the expected behavior?

Reply:
yes, ref to https://github.com/pingcap/tidb/blob/c47f4c914624371442ce49bf42956496c1593963/pkg/expression/builtin_json.go#L1774-L1776

Reply:
waiting for https://github.com/pingcap/tidb/pull/50933

Reply:
fixed.